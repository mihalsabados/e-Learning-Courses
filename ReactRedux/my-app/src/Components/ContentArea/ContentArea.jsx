import React  from 'react';
import "./ContentArea.css"
import { useSelector } from "react-redux";
import { useEffect, useState } from "react";

export default function ContentArea() {
    const courses = useSelector(state => state.courses)
    const [courseGroups, setCourseGroups] = useState({});

    useEffect(() => {
        setCourseGroups(groupBy(courses));
    }, [])


    const groupBy = (courses) =>{
        return courses.reduce((result, item) => {
            const catKey = item['category'];
            if (!result[catKey]) {
              result[catKey] = [];
            }
            result[catKey].push(item);
            return result;
          }, {});
    }

    const displayDate = (date) => {
        let year = date.getFullYear();
        let month = date.toLocaleString('default', { month: 'long' });
        let day = date.getDate().toString().padStart(2, '0');
        return month+" "+day+", "+year;
    }

    return (
    <div className="courses-area">
        <div className="courses-header">
            <h2>e-Learning Courses</h2>
        </div>
        {courseGroups && Object.keys(courseGroups).map((category, id)=>{
            const imageSrc = courseGroups[category][0]['imageSrc']

            return (
            <div className="course-group" key={id}>
                <div className="course-img">
                    <img src={imageSrc} alt={category}/>
                </div>
                <div className="course-list">

                    {courseGroups[category].map((course, cId)=>{
                        const circleArray = ["circle", "circle", "circle", "circle", "circle"].map(
                            (val, i)=>{
                                let elem = val;
                                if(course.progress >= i+1)
                                    elem+=" full";
                                return elem;
                            });
                        
                        return (
                        <div className="course" key={cId}>
                            <div className="course-text">
                                <p>{course.title}</p>
                            </div>
                            <div className="course-progress">
                                <div className="course-duration">
                                    <p>{course.duration}min</p>
                                    <p>To be done by:</p>
                                </div>
                                <div className="course-indicators">
                                    <div className="course-circles">
                                        {circleArray.map((circleClass, cirId)=>(
                                            <div className={circleClass} key={cirId}></div>
                                        ))}
                                    </div>
                                    <p className="done-by">{course.endDate?displayDate(course.endDate):"-"}</p>
                                </div>
                            </div>

                        </div>
                        )
                    })}
                </div>
            </div>
            )
        })}

    </div>
    );
}