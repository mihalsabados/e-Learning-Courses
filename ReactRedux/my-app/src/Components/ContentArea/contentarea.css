.courses-area {
	margin-left: 30px;
	width: 75%;
	margin-bottom: 50px;
}

.courses-header {
	margin-top: 40px;
}

.courses-header h2 {
	font-weight: 400;
	color: #666973;
}

.course-group {
	background-color: white;
	display: flex;
	margin-bottom: 20px;
}

.course-img,
.course-img img {
	height: 50px;
	margin: 13px;
}

.course-list {
	margin-block: 15px;
	width: 100%;
	padding-right: 5%;
}

.course {
	display: flex;
	width: 100%;
}

.course-text {
	width: 70%;
}

.course-text p {
	font-weight: 600;
	font-size: 19px;
	color: #666973;
}

.course-progress {
	display: flex;
	margin-bottom: 0;
	margin-left: 8%;
	color: #666973;
}

.course-duration {
	margin-top: 13px;
}

.course-duration p {
	padding-block: 0;
	margin-block: 5px;
	text-wrap: nowrap;
}

.course-circles {
	margin-top: 17px;
	display: flex;
}

.circle {
	height: 15px;
	width: 15px;
	border-radius: 50%;
	background-color: #d8d8d8;
	margin-inline: 4px;
}

.full {
	background-color: #fddd5b;
}

.course-restriction p {
	margin-top: 0;
}

.done-by {
	padding-left: 3px;
	margin-top: 8px;
}

.course-indicators {
	margin-left: 3px;
}

@media only screen and (max-width: 1000px) {
	.courses-area {
		margin-left: 0;
		width: 100%;
	}
}

@media only screen and (max-width: 800px) {
	.course {
		display: block;
	}
	.course-progress {
		margin: 0;
	}
	.course-text p {
		margin-bottom: 0;
	}
	.course-duration {
		margin-top: 5px;
	}
	.course-circles {
		margin-top: 7px;
	}
}
