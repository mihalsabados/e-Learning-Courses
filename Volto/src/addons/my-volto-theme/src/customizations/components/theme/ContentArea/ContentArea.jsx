import React from 'react';
import { useSelector } from 'react-redux';

export default function ContentArea() {
  // Load courseCategories from redux store
  const courseCategories = useSelector((state) => state.courses);

  //Display date in 'MMMM dd, yyyy' format
  const displayDate = (date) => {
    let year = date.getFullYear();
    let month = date.toLocaleString('default', { month: 'long' });
    let day = date.getDate().toString().padStart(2, '0');
    return month + ' ' + day + ', ' + year;
  };

  // create array for circles className depending on courseProgress, all divs have 'circle' className and full circles have added 'full'
  const getCircleClassNames = (courseProgress) => {
    return ['circle', 'circle', 'circle', 'circle', 'circle'].map((val, i) => {
      let elem = val;
      if (courseProgress >= i + 1) elem += ' full';
      return elem;
    });
  };

  return (
    <div className="courses-area">
      <div className="courses-header">
        <h2>e-Learning Courses</h2>
      </div>
      {/* Map through course categories */}
      {courseCategories.items &&
        courseCategories.items.map((category, id) => {
          const imageSrc = category.imageSrc;

          return (
            <div className="course-group" key={id}>
              <div className="course-img">
                <img src={imageSrc} alt={category} />
              </div>
              <div className="course-list">
                {/* Map through courses in categories */}
                {category.courses.map((course, cId) => {
                  const circleArray = getCircleClassNames(course.progress);

                  return (
                    <div className="course" key={cId}>
                      <div className="course-text">
                        <p>{course.title}</p>
                      </div>
                      <div className="course-progress">
                        <div className="course-duration">
                          <p>{course.duration}min</p>
                          <p>To be done by:</p>
                        </div>
                        <div className="course-indicators">
                          <div className="course-circles">
                            {/* Map throught array for circles className */}
                            {circleArray.map((circleClass, cirId) => (
                              <div className={circleClass} key={cirId}></div>
                            ))}
                          </div>
                          <p className="done-by">
                            {course.endDate ? displayDate(course.endDate) : '-'}
                          </p>
                        </div>
                      </div>
                    </div>
                  );
                })}
              </div>
            </div>
          );
        })}
    </div>
  );
}
