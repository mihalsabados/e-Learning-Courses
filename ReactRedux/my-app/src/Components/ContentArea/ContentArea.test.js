/* eslint-disable no-undef */
import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import ContentArea from "./ContentArea";
import { store } from "../../store";
import renderer from "react-test-renderer";

test("ContentArea component should render", () => {
	render(
		<Provider store={store}>
			<ContentArea />
		</Provider>
	);

	expect(screen.getByText("e-Learning Courses")).toBeInTheDocument();

	expect(
		screen.getByText(
			"Intensification of money laundering prevention - private customer advice"
		)
	).toBeInTheDocument();
	expect(
		screen.getByText(
			"Basic Training - Prevention of Money Laundering and Terrorist Financing (V3.0)"
		)
	).toBeInTheDocument();
	expect(
		screen.getByText(
			"Advanced Training - Prevention of Money Laundering and Terrorist Financing (V3.0)"
		)
	).toBeInTheDocument();

	expect(screen.getByText("IT security basics (V1.01.3)")).toBeInTheDocument();
	expect(
		screen.getByText(
			"Basic Training IT security and information security (V1.0)"
		)
	).toBeInTheDocument();
	expect(
		screen.getByText(
			"Advanced Training IT Security and Information Security (V1.1)"
		)
	).toBeInTheDocument();

	expect(
		screen.getByText("Basic training data protection (V2.0)")
	).toBeInTheDocument();
	expect(screen.getByText("Privacy Basics (V1.00.2)")).toBeInTheDocument();
	expect(
		screen.getByText("In-depth training on data protection (V1.0)")
	).toBeInTheDocument();
	expect(screen.getByText("Privacy Basics 1.0")).toBeInTheDocument();

	expect(
		screen.getByText("WpHG compliance basics (V1.00.3)")
	).toBeInTheDocument();
	expect(
		screen.getByText("In-depth WpHG - sales representative (V1.00.2)")
	).toBeInTheDocument();
});

test("ContentArea component snapshot", () => {
	const component = renderer.create(
		<Provider store={store}>
			<ContentArea />
		</Provider>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
