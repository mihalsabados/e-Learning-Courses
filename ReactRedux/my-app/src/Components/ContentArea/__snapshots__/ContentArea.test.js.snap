// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentArea component snapshot 1`] = `
<div
  className="courses-area"
>
  <div
    className="courses-header"
  >
    <h2>
      e-Learning Courses
    </h2>
  </div>
</div>
`;
